lib/Page.js

```diff
@@ -87,6 +89,30 @@ class Page extends EventEmitter {
 
     this._screenshotTaskQueue = screenshotTaskQueue;
 
+    /** @type {!Map<string, Worker>} */
+    this._workers = new Map();
+    client.on('Target.attachedToTarget', event => {
+      if (event.targetInfo.type !== 'worker') {
+        // If we don't detach from service workers, they will never die.
+        client.send('Target.detachFromTarget', {
+          sessionId: event.sessionId
+        });
+        return;
+      }
+      const session = client._createSession(event.targetInfo.targetId, event.sessionId);
```

Comment:
should this be a part of `CDPSession`?

Reply:
I don't think so. If people are already interacting with raw protocol, they can handle sub-targets without our help.